### PR TITLE
qemu: fix strictDeps

### PR DESCRIPTION
**Pull request by ofalvai**
Ref: https://github.com/NixOS/nixpkgs/issues/178468

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchurl, fetchpatch, python3Packages, zlib, pkg-config, glib, overrideSDK, buildPackages
 , pixman, vde2, alsa-lib, flex, pcre2
-, bison, lzo, snappy, libaio, libtasn1, gnutls, nettle, curl, dtc, ninja, meson
+, bison, lzo, snappy, libaio, libtasn1, gnutls, nettle, curl, dtc, ninja, meson, perl
```
Comment (SigmaSquadron) [inline pkgs/applications/virtualization/qemu/default.nix:3]:
```suggestion
, bison, lzo, snappy, libaio, libtasn1, gnutls, nettle, curl, dtc, ninja, meson
```

I don't follow; why are we adding `perl` when the package works fine without it?

**Review by SigmaSquadron — COMMENTED** [review]
Why is this going to `staging`?

File: pkgs/applications/virtualization/qemu/default.nix (hunk 1 above)
Comment (ofalvai) [reply pkgs/applications/virtualization/qemu/default.nix:3]: I caught this problem by settings `nixpkgs.config.strictDepsByDefault` to true. There is more info in this issue: https://github.com/NixOS/nixpkgs/issues/178468

For why this happens to qemu specifically, my assumption is one of its dependencies propagates buildInputs/nativeBuildInputs incorrectly, but it's not surfaced when `strictDeps` is false.

**Comment by ofalvai** [comment]
> Why is this going to staging?

I wasn't sure how many rebuilds this will cause, so I chose staging. It's not an urgent fix so I'm fine with the staging merge.

**Review by SigmaSquadron — APPROVED** [review]
Thanks for answering my questions; LGTM.

**Review by alyssais — APPROVED** [review]
Thanks!

(I've switched base to master, because it's good to limit staging to only changes that need to be there, to avoid making it more out of sync and difficult to merge than necessary.)

**Pull request merged**